src/ringcentral-web-phone.js

```diff
@@ -1155,6 +1177,226 @@
+        options = options || {};
+
+        var networkType = calculateNetworkUsage(qosStatsObj) || '';
+        var targetUrl = options.targetUrl || 'rtcpxr@rtcpxr.ringcentral.com:5060';
+        var event = options.event || 'vq-rtcpxr';
+        options.expires = 60;
+        options.contentType = "application/vq-rtcpxr";
+        options.extraHeaders = options.extraHeaders || [];
+        options.extraHeaders.push('p-rc-client-info:' + 'cpuRC=0:0;cpuOS=0:0;netType=' + networkType + ';ram=0:0');
+
+        var calculatedStatsObj =  calculateStats(qosStatsObj);
+        var body = createPublishBody(calculatedStatsObj);
+        var pub = session.ua.publish(targetUrl, event, body, options);
+        qosStatsObj.status = false;
+        pub.close();
+        session.emit('qos-published',body);
+    }
+
+    /*--------------------------------------------------------------------------------------------------------------------*/
+
+    /**
+     * @this {SIP.Session}
+     */
+    function startQosStatsCollection(){
+
+        var session =  this;
+        var qosStatsObj = getQoSStatsTemplate();
+
+        qosStatsObj.callID = session.request.call_id||'';
+        qosStatsObj.fromTag  = session.from_tag || '';
+        qosStatsObj.toTag  = session.to_tag || '';
+        qosStatsObj.localID = session.request.headers.From[0].raw||session.request.headers.From[0];
+        qosStatsObj.remoteID = session.request.headers.To[0].raw||session.request.headers.To[0];
+        qosStatsObj.origID = session.request.headers.From[0].raw||session.request.headers.From[0];
+
+
+        var repeatInterval = session.ua.qosCollectInterval;
+        var peer =  session.sessionDescriptionHandler.peerConnection;
+        var previousGetStatsResult;
+
+        getStats(peer, function (getStatsResult){
+            previousGetStatsResult = getStatsResult;
+            qosStatsObj.status =  true;
+            var network = getNetworkType(previousGetStatsResult.connectionType);
+            qosStatsObj.localAddr = previousGetStatsResult.connectionType.local.ipAddress[0];
+            qosStatsObj.remoteAddr = previousGetStatsResult.connectionType.remote.ipAddress[0];
+            previousGetStatsResult.results.forEach(function (item) {
+                if (item.type === 'ssrc' && item.transportId === 'Channel-audio-1' && item.id.includes('recv')) {
+                    qosStatsObj.jitterBufferDiscardRate = item.googSecondaryDiscardedRate || 0;
+                    qosStatsObj.packetLost = item.packetsLost;
+                    qosStatsObj.packetsReceived = item.packetsReceived;
+                    qosStatsObj.totalSumJitter += parseFloat(item.googJitterBufferMs);
+                    qosStatsObj.totalIntervalCount += 1;
+                    qosStatsObj.JBM = Math.max(qosStatsObj.JBM, parseFloat(item.googJitterBufferMs));
+                    qosStatsObj.netType = addToMap(qosStatsObj.netType, network);
+                }
+            });
+        }, repeatInterval);
```

Comment:
No need for the `repeatInterval` var, use `session.ua.qosCollectInterval` directly